java/arcs/core/storage/driver/Database.kt

```diff
@@ -238,6 +239,7 @@ class DatabaseDriver<Data : Any>(
             is DatabaseData.Entity -> data.rawEntity.toCrdtEntityData(data.versionMap) {
                 when (it) {
                     is Reference -> it
+                    is RawEntity -> CrdtEntity.Reference.wrapReferencable(it)
```

Comment:
It isn't immediately apparent to a reader that this is for the inline nested entities case. Is it worth a comment, or, is it not because this could also be applicable to other types?

Reply:
I want to figure out how to do this better in general; there are several of these clauses spread throughout the codebase and missing one doesn't always immediately produce errors.

@jblebrun @jasonwyatt any suggestions?

Reply:
A comment would be good. I don't really understand what the difference is between wrapReferencable and buildReference without digging into them further. 

It seems like this when-expression could be factored into its own function, living on CrdtEntity.Reference maybe?

Reply:
I think so; but I don't understand why it wasn't done that way in the first place. In particular, I'm concerned about potential build ramifications.

Reply:
This type of transformation should probably be extracted into an interface from which Reference/RawEntity/ReferencableList/etc would extend.

Reply:
At a glance, it appears an easy step in the right direction is moving the nested `when` inside of `RawEntity.toCrdtEntityData`